packages/ics721/src/helpers.rs

```diff
@@ -147,8 +151,13 @@ pub fn get_instantiate2_address(
     let CodeInfoResponse { checksum, .. } = deps.querier.query_wasm_code_info(code_id)?;
 
     let canonical_cw721_addr = instantiate2_address(&checksum, &canonical_creator, salt)?;
-
-    Ok(deps.api.addr_humanize(&canonical_cw721_addr)?)
+    if let Some(len) = CONTRACT_ADDR_LENGTH.may_load(deps.storage)? {
+        Ok(deps
+            .api
+            .addr_humanize(&canonical_cw721_addr[..len].into())?)
+    } else {
+        Ok(deps.api.addr_humanize(&canonical_cw721_addr)?)
+    }
```

Comment:
Here is the main logic for slicing addr

Reply:
@humanalgorithm look here, for Injective `CONTRACT_ADDR_LENGTH` is set with `20`. All other chains have no entry in `CONTRACT_ADDR_LENGTH` storage.

Value is then used for slicing first 20 bytes: `canonical_cw721_addr[..len]`